### DISCUSSION TITLE
Hubspot source check fail

### DISCUSSION THREAD

**Discussion by taticorrea-unico** [airbytehq/airbyte#37798] [Connector Questions]
Airbyte: Version 0.50.29
Source: Hubspot 4.1.2 (latest)
Destination: BigQuery 2.3.30

I'm trying to create a new connection to the hubspot source but the verification step takes about 5 minutes and gives the following error
`Server temporarily unavailable (http.502XXXXXXX). `

Can you help me?

**Comment by NAjustin** [comment]
I would recommend checking the timeouts—many platforms have default timeouts on the load balancer or HTTP proxy as low as 30 seconds, which will result in a 502 gateway error on any operation that takes longer than that. Since connection checks and syncs have to spin up the pods, then run them, it is easy for it to exceed lower limits. In this case I'm guessing that limit is 300 seconds (5 minutes).

I would recommend setting this to something like 600, which has worked for us on all our HubSpot connections. You can also set it arbitrarily higher.

The specifics of where and how vary by hosting platform and how you're deploying Airbyte (e.g. Docker vs. Helm/Kubernetes). Including those details will help you get better, more-detailed suggestions.

**Reply by taticorrea-unico** [reply]
My airbyte deploy is on k8s with heml chart

**Reply by taticorrea-unico** [reply]
where i can set this timeout ? In airbyte_values file?

**Reply by NAjustin** [reply]
@taticorrea-unico What platform are you deployed on (e.g. AWS, GCP, etc.)? Usually the timeouts you're interested in are going to be on the load balancer, but depends on how you're deploying things. It could also be on a proxy or gateway of some other sort. So usually these are things you need to adjust at the infrastructure layer, and can't always set in `values.yaml` (although some platforms allow you to add annotations to the `service` or `ingress` sections to pass timeout config through)

**Reply by LucLagrange** [reply]
Hello :) 

I had a similar issue (502 error, Airbyte deployed on K8S with an ingress) and increasing the timeout did work for me.

**Comment by alexanderNthl** [comment]
I fixed this issue by increasing the HTTP_IDLE_TIMEOUT of the server. If you run the community version locally, this can be achieved via:

1.  Edit existing variable
    `kubectl edit -n airbyte-abctl configmap airbyte-abctl-airbyte-env`
    `HTTP_IDLE_TIMEOUT: 600s`
2. Restart server deployment
    `kubectl rollout restart deployment airbyte-abctl-server -n airbyte-abctl`
